### ISSUE TITLE
How to serialize dns solver requests

### ISSUE THREAD

**Issue by KnicKnic** [xenolf/lego#680]
I am using lego through traefik. The provider I am using is DuckDns. 
There is a imitation with duckdns that it only has one txt record per account, that txt record will be presented for any lookup request on any subdomain.

So `test1.test.duckdns.org` and `test2.test.duckdns.org` do not have different txt records. Currently I want to be able to request in traefik 
```toml
 [[acme.domains]]
   main = "test1.test.duckdns.org"
 [[acme.domains]]
   main = "test2.test.duckdns.org"
```
I was wondering how I could serialize the requests such that the two requests do not overlap and that `test1` is validated and then `test2` is validated (currently `test2` will succeed and then eventually `test1` will get retried). I am okay with all requests going through DuckDNS are serialized, but if possible it would be nice to limit it only to one per account. I am curious to this interaction across the operation that spans both `Present` and `CleanUp`.

@ldez - due to already paging in context on https://github.com/xenolf/lego/pull/676

Thanks!

**Comment by ldez** [comment]
I saw the problem with DuckDNS tomorrow, and I may have a solution.
For the moment, I have to think about it.

**Comment by ldez** [comment]
Closed by #700
